### PR TITLE
Geotransolver Model

### PR DESCRIPTION
**Pull request by coreyjadams**
# PhysicsNeMo Pull Request

This PR brings GeoTransolver to physicsnemo and unifies the training recipe with Transolver.

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: examples/cfd/external_aerodynamics/transformer_models/src/inference_on_vtk.py [hunk 1]
```diff
@@ -0,0 +1,729 @@
...
+                stream_velocity=stream_velocity,
+                run_idx=run_idx,
+            )
+
+            # Process through datapipe (adds batch dimension)
+            batch = datapipe(data_dict)
+
+            # Run batched inference using imported function from inference_on_zarr
+            with torch.no_grad():
+                _, _, (predictions, _) = batched_inference_loop(
```
Comment (RishikeshRanade) [inline examples/cfd/external_aerodynamics/transformer_models/src/inference_on_vtk.py:641]: There is a shape mismatch here in the global_parameters setting. Global parameters have an extra dimension (1, 1, 2) in this case. We need to modify the datapipe to squeeze the first dimension.

File: examples/cfd/external_aerodynamics/transformer_models/README.md [hunk 2]
```diff
@@ -1,134 +1,74 @@
...
-## External Aerodynamics CFD Example: Overview
+### Transolver
+
+`Transolver` is a high-performance surrogate model for CFD solvers. The Transolver model adapts the Attention mechanism, encouraging the learning of meaningful representations. In each PhysicsAttention layer, input points are projected onto state vectors through learnable transformations and weights. These transformations are then used to compute self-attention among all state vectors, and the same weights are reused to project states back to each input point.
+
+By stacking multiple PhysicsAttention layers, the `Transolver` model learns to map from the functional input space to the output space with high fidelity. The PhysicsNeMo implementation closely follows the original Transolver architecture ([https://github.com/thuml/Transolver](https://github.com/thuml/Transolver)), but introduces modifications for improved numerical stability and compatibility with NVIDIA TransformerEngine.
 
-This directory contains the essential components for training and evaluating a
-model tailored to external aerodynamics CFD problems built on `Transolver`.
+### Typhon
```
Comment (RishikeshRanade) [inline examples/cfd/external_aerodynamics/transformer_models/README.md]: This needs to be renamed to GeoTransolver

**Review by RishikeshRanade — APPROVED** [review]

File: examples/cfd/external_aerodynamics/transformer_models/src/inference_on_vtk.py (hunk 1 above)
Comment (coreyjadams) [reply examples/cfd/external_aerodynamics/transformer_models/src/inference_on_vtk.py:641]: Gonna get this in the next update.
